packages/core/src/components/axes/tool-bar.ts

```diff
@@ -0,0 +1,325 @@
+			);
+
+			const hasOpenedOverflowMenuOptions =
+				this.overflowMenuOptions
+					.selectAll("ul.bx--overflow-menu-options--open")
+					.size() > 0;
+
+			document.body.addEventListener("click", function () {
+				// always clear menu icon hover state
+				self.setOverflowMenuIconHover(false);
+				self.services.events.dispatchEvent(
+					Events.Toolbar.HIDE_MENU_ITEMS
+				);
+			});
+
+			if (hasOpenedOverflowMenuOptions) {
+				this.setOverflowMenuIconHover(true);
+				this.overflowMenuOptions.html(this.getOverflowMenuHTML());
+			} else {
+				this.setOverflowMenuIconHover(false);
+				this.overflowMenuOptions.html(null);
+			}
+		}
+	}
+
+	getOverflowMenuHTML() {
+		// supports only reset zoom for now
+		// if getResetZoomMenuItem() return nothing
+		// don't render whole overflow menu
+		if (!this.getResetZoomMenuItem()) {
+			return "";
+		}
+
+		let defaultHTML;
+		defaultHTML = `<div data-floating-menu-container="true"
+			data-floating-menu-direction="bottom" role="main">
+			<ul class="bx--overflow-menu-options bx--overflow-menu--flip bx--overflow-menu-options--open"
+				tabindex="-1" role="menu" aria-label="Menu" data-floating-menu-direction="bottom"
+				style="left:${this.overflowMenuX}px; top:${this.overflowMenuY}px;">`;
+		defaultHTML += this.getResetZoomMenuItem();
+		defaultHTML += `</ul></div>`;
+
+		return defaultHTML;
+	}
+
+	getResetZoomMenuItem() {
+		const resetZoomOption = Tools.getProperty(
+			this.model.getOptions(),
+			"toolBar",
+			"toolBarMenuItems",
+			"resetZoom"
+		);
+		if (!resetZoomOption.enabled || !this.isZoomBarEnabled) {
+			return "";
+		} else {
+			return `<li
+						class="bx--overflow-menu-options__option">
+						<button class="bx--overflow-menu-options__btn" role="menuitem"  title="Reset"
+							data-floating-menu-primary-focus
+							id="${this.resetZoomMenuItemId}">
+							<div class="bx--overflow-menu-options__option-content">
+								${resetZoomOption.text}
+							</div>
+						</button>
+					</li>`;
+		}
+	}
+
+	handleOverflowMenuEvent() {
+		if (
+			this.overflowMenuOptions
+				.selectAll("ul.bx--overflow-menu-options--open")
+				.size() > 0
+		) {
+			// hide overflow menu
+			this.setOverflowMenuIconHover(false);
+			this.services.events.dispatchEvent(Events.Toolbar.HIDE_MENU_ITEMS);
+		} else {
+			// show overflow menu
+			this.setOverflowMenuIconHover(true);
+			this.services.events.dispatchEvent(Events.Toolbar.SHOW_MENU_ITEMS);
+			const self = this;
+			const resetZoomButtonElement = document.getElementById(
+				this.resetZoomMenuItemId
+			);
+			if (resetZoomButtonElement !== null) {
+				resetZoomButtonElement.addEventListener(
+					"click",
+					function () {
+						self.zoomService.resetZoomDomain();
+
+						self.setOverflowMenuIconHover(false);
+
+						self.services.events.dispatchEvent(
+							Events.Toolbar.HIDE_MENU_ITEMS
+						);
+					},
+					true
+				);
+			}
+		}
+		event.stopImmediatePropagation();
+	}
+
+	setOverflowMenuIconHover(enable) {
```

Comment:
this shouldn't really be a method, it's unnecessary. also rather than doing `"icon-overflowRect"` & `"icon-overflowRect"` we should do `"icon-overflowRect"` & `"icon-overflowRect hovered"`

Reply:
Updated in https://github.com/carbon-design-system/carbon-charts/pull/717/commits/ad2c4fbc6a1e14980a1320ae416bb6ce08691bde.